tests/integration/strings_test.go

```diff
@@ -179,7 +179,7 @@ var _ = Context("strings", func() {
 			shouldRender("{{ ';'.join(['s', 'b', '3']) }}", "s;b;3")
 		})
 		Context("format", func() {
-			shouldRender("{{ 'foo={:,=-10.5G}'.format(\"77.11121111111112\") }}", "foo=,,,,77.111")
+			shouldRender("{{ 'foo={:,=-10.5G}'.format(77.11121111111112) }}", "foo=,,,,77.111")
```

Comment:
This can be supported if select the default dialect to enable [type juggling](https://github.com/NikolaLohinski/gonja/pull/15/files#diff-d3c51092aaa2c7643a54a9630b03e16dc4b2e39f9556d34341c9fa8c7dd2a047R20) but i'm not sure we want that..

Reply:
Nah, if it's a beta feature in python, I'd rather not enable it in here by default

Reply:
Not even a beta feature in python. Not present at all in python. The rationale is that if not only the template strings but also values are sourced from the user; it is likely that all values will be of string type; as such we can be forgiving in accepting strings, that represents a number, as a number. 

I agree that it should be left out of the standard option and rather let consumers opt in if they want this feature. The biggest issue i have with this approach is that i don't see an easy way to provide this config to a single gonja instance that get's passed down the the PyString.Format() so for now the only option to allow type juggling would be to override `DefaultDialect` which would be a global setting. 

Would love for this to be encapsulated in some instantiated version of gonja but that seems like a heavy task.

Reply:
Ok gotcha. Thanks for the super clear explanation.

In this case, we should stick to what Python is offering. We don't want to assume the intent of the library user by being smarter in type casting. An invalid input should be invalid, especially in a "strongly" typed language such as Go. IMHO the library shouldn't go further than Python's behavior, as we're trying to replicate it to support non-pure-jinja templates. A template that leverages that type juggling wouldn't work in Python's Jinja, which is not what we want.

Makes sense ?